pyomo/contrib/mcpp/pyomo_mcpp.py

```diff
@@ -257,13 +278,15 @@ def register_var(self, var, lb, ub):
                 % (var.name, ub))
         if var_val is None:
             var_val = (lb + ub) / 2
-            logger.warning(
-                'Var %s missing value. Assuming midpoint value of %s'
-                % (var.name, var_val))
-        return self.mcpp.new_createVar(
+            self.missing_value_warnings.append(
+                'Var %s missing value. Assuming midpoint value of %s' % (var.name, var_val))
+        return self.mcpp.newVar(
             lb, var_val, ub, self.num_vars, var_idx)
 
     def finalizeResult(self, node_result):
+        for r in self.refs:
+            if r != node_result:
+                self.mcpp.release(r)
         return node_result
```

Comment:
While this cleans up all the intermediate nodes, I am not sure that the root node (that we return here) ever gets released.

Reply:
I added it to the `__del__` method of the visitor object.